File: internal/sem/testdata/langmatrix/lua/ledger.lua
local Ledger = {}
Ledger.__index = Ledger

function Ledger.new()
  return setmetatable({ total = 0 }, Ledger)
end

function Ledger.add(self, amount)
  return self.total + amount
end

local function ledger_double(amount)
  local ledger = Ledger.new()
  return Ledger.add(ledger, amount) * 2
end

return ledger_double

File: internal/sem/testdata/langmatrix/perl/Ledger.pl
package Ledger;

sub new {
    my ($class) = @_;
    return bless { total => 0 }, $class;
}

sub add {
    my ($self, $amount) = @_;
    return $self->{total} + $amount;
}

package main;

sub ledger_double {
    my ($amount) = @_;
    my $ledger = Ledger->new();
    return $ledger->add($amount) * 2;
}

1;
